examples/how_to_benchmark/tutorial_6_within_session_splitter.py

```diff
@@ -0,0 +1,211 @@
+# which supports three levels of model generalization:
+#
+#     - Within-session: test generalization across trials within a single session
+#     - Cross-session: test generalization across different recording sessions
+#     - Cross-subject: test generalization across different brains
+#
+# Each level decreases in specialization, moving from highly subject-specific models
+# to those that can generalize across individuals.
+#
+# This tutorial focuses on within-session evaluation to establish a reliable
+# baseline for model performance before attempting more challenging generalization tasks.
+
+import warnings
+
+import matplotlib.pyplot as plt
+
+# Standard imports
+import pandas as pd
+import seaborn as sns
+
+# MNE + sklearn for pipeline
+from mne.decoding import CSP
+from sklearn.discriminant_analysis import LinearDiscriminantAnalysis as LDA
+from sklearn.pipeline import make_pipeline
+
+import moabb
+
+# MOABB components
+from moabb.datasets import BNCI2014_001
+from moabb.evaluations.splitters import WithinSessionSplitter
+from moabb.paradigms import LeftRightImagery
+
+
+# Suppress warnings and enable informative logging
+warnings.filterwarnings("ignore")
+moabb.set_log_level("info")
+
+###############################################################################
+#  Load the dataset and paradigm
+###############################################################################
+# We use the BNCI2014_001 dataset: BCI Comp IV dataset 2a (motor imagery)
+dataset = BNCI2014_001()
+# Restrict to a few subjects to keep runtime reasonable for demonstration
+dataset.subject_list = [1, 2, 3]
+
+# Define the paradigm: here, left vs right hand imagery
+paradigm = LeftRightImagery()
+
+###############################################################################
+#  Extract data: epochs (X), labels (y), and trial metadata (meta)
+###############################################################################
+# This call downloads (if needed), preprocesses, epochs, and labels the data
+X, y, meta = paradigm.get_data(dataset=dataset, subjects=dataset.subject_list)
+
+# Inspect the shapes: X is trials × channels × timepoints; y is labels; meta is info
+print("X shape (trials, channels, timepoints):", X.shape)
+print("y shape (trials,):", y.shape)
+print("meta shape (trials, info columns):", meta.shape)
+print(meta.head())  # shows subject/session for each trial
+
+###############################################################################
+#  Build a classification pipeline: CSP to LDA
+###############################################################################
+# CSP finds spatial filters that maximize variance difference between classes
+# LDA is a simple linear classifier on the CSP features
+pipe = make_pipeline(
+    CSP(n_components=6, reg=None),  # reduce to 6 CSP components
+    LDA(),  # classify based on these features
+)
+print("Pipeline steps:", pipe.named_steps)
```

Comment:
@bruAristimunha's comment is this:
```suggestion
pipe
```

Reply:
Sorry this was already resolved